src/Place/Place.php

```diff
@@ -222,6 +229,28 @@ protected function applyAddressTranslated(AddressTranslated $addressTranslated)
         $this->addresses[$addressTranslated->getLanguage()->getCode()] = $addressTranslated->getAddress();
     }
 
+    public function markAsDuplicateOf(string $placeIdOfCanonical): void
+    {
+        if ($this->isDeleted) {
+            throw CannotMarkPlaceAsDuplicate::becauseItIsDeleted($this->placeId);
+        }
+
+        if ($this->isDuplicate) {
+            throw CannotMarkPlaceAsDuplicate::becauseItIsAlreadyADuplicate($this->placeId);
+        }
+
+        $this->apply(new MarkedAsDuplicate($this->placeId, $placeIdOfCanonical));
+    }
+
+    public function markAsCanonicalFor(string $placeIdOfDuplicate): void
```

Comment:
This method should also check that the canonical place is not a duplicate of another one, right?

Reply:
Yes, looks like that's still missing. Will add.